lib/src/receipt_proof.nr

```diff
@@ -0,0 +1,82 @@
+use crate::{
+    trie_proof::TrieProof,
+    const::{MAX_TRIE_NODE_LENGTH, LEAF, HASH_LENGTH, HASH_NIBBLE_LENGTH, RLP_IDX_KEY_NIBBLE_LENGTH, MAX_RECEIPT_RLP_LENGTH},
+    utils::{memcpy, byte_value, assert_subarray},
+    node::{key_as_nibbles, verify_internal_node, verify_node_hash, resolve2}
+};
+use dep::std::hash::keccak256;
+use crate::rlp;
+
+/// Type alias for receipt proofs. Assumes value is left-padded with zeros.
+type ReceiptProof<PROOF_LEN, RECEIPT_RLP_LENGTH> = TrieProof<3, PROOF_LEN, RECEIPT_RLP_LENGTH>;
```

Comment:
We could generalize type, as it will be the same for Receipts, Transactions, Withdraws and perhaps blobs.
```rust
type TrieProof<PROOF_LEN, RECEIPT_RLP_LENGTH> = HashedKeyTrieProof<3, PROOF_LEN, RECEIPT_RLP_LENGTH>;

type ReceiptProof<PROOF_LEN> = TrieProof<3, PROOF_LEN, RECEIPT_RLP_LENGTH>;
type TransactionProof<PROOF_LEN> = TrieProof<3, PROOF_LEN, TRANSACTION_RLP_LENGTH>;
```

Alternativelly, to avoid renaming main type in library:
```rust
type KeyValueTrieProof<PROOF_LEN, RECEIPT_RLP_LENGTH> = TrieProof<3, PROOF_LEN, RECEIPT_RLP_LENGTH>;

```

Cound be done in next PR.